### ISSUE TITLE
Bug: [array-type] doesn't handle generic params/local types named `Array`

### ISSUE THREAD

**Issue by paperclover** [typescript-eslint/typescript-eslint#11964]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=C4TwDgpgBA4hB2EBOBLAxgHgIJKQQxCggA9gEATAZygFd4BreAewHd4BtAXQD4oBeKAG8oeXAQBcUHPkIBfANwAoReQhoANqOgBbJuRrroAcgBMRoVEVQooSFLEgMATV4CnS62ibxKwKCElpAgwTbiVZRSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AQ2mncNqNPRREXAPbRI4MAF8QUoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
type Generic<Array extends unknown[]> = { array: Array };

// secondary example
declare module '2' { 
  type Array<Y> = Y;
  const y: Array<2>;
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/array-type": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Pass

### Actual Result

> Array type using 'Array<any>' is forbidden. Use 'any[]' instead. 1:50 - 1:55

**Comment by SungHyun627** [comment]
it seems to be caused by missing scope checks for the 'Array' identifier in array-type.ts. May I take this on? @JoshuaKGoldberg @kirkwaiblinger

**Comment by JoshuaKGoldberg** [comment]
We appreciate the enthusiasm ❤️ but there's no need to tag individual team members. Once an issue is marked as `accepting prs` then anybody can tackle it (https://typescript-eslint.io/contributing/issues#issue-claiming).

The team's been swamped this month and hasn't had time to triage this issue yet. We'll try to get to it Soon ™️!

**Comment by kirkwaiblinger** [comment]
duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/11287

**Comment by paperclover** [comment]
> The reason we don't do anything more expensive or heavier is because it's widely considered bad practice to shadow well-known global names like this.

this is annoying because the rule auto fixes valid code into failing code. though the failing code can get caught by other rules, it is extremely unsatisfying to see this closed duplicate of a `wontfix`.

**Comment by paperclover** [comment]
actually. this would be 99% solved if the autofixer didn't apply for `Array` without a generic application, since the actual `Array` type has a required generic parameter. this would solve the first case in my example. right now it converts it to `any[]` which is insane.

**Comment by kirkwaiblinger** [comment]
We discussed internally and figured we can accept a PR for this. Reopening!
